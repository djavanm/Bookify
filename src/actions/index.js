export const search = query => ({
  type: 'SEARCH_BOOKS',
  query
});
