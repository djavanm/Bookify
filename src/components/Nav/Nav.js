import React from 'react';
import { Link } from 'react-router-dom';

const Nav = ({ user }) => {
  return (
    <nav>
    <h1>Bookify</h1>
    { !user && <Link to='/login'><button> SIGN IN </button></Link> }
    { user && <button> Show Favorites </button> }
    { user && <h2> { user.name } </h2> }
    </nav>
  )
}

export default Nav;
