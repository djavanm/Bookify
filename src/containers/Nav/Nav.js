import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser, showGenreFilter, showAllFilter} from '../../actions';
import { bindActionCreators } from 'redux';
import { FaLeaf } from "react-icons/fa";

export const Nav = ({ currentUser, logoutUser, home, genres, showGenreFilter, showAllFilter }) => {
  const currentGenres = genres.map((genre, index) => {
  return <button className='nav-btn' key={index+1} onClick={() => showGenreFilter(genre)}>{genre}</button>
  });
  const showAll = <button className='nav-btn' key={0} onClick={() => showAllFilter()}>Show All</button>;
  const allGenres = [showAll, ...currentGenres];
  return (
    <>
    <nav>
      <h1 className='header-title'>Bookify <FaLeaf color='#07804B' size={40} /> </h1>
      <section className='nav-buttons'>
      { !currentUser && <Link to='/login'><button className='nav-btn'> SIGN IN </button></Link> }
      {currentUser && <h2 className='nav-welcome'> Welcome,<span className="user-name"> {currentUser.name}</span>  </h2> }
      { currentUser && !home && <Link className='link' to='/my-collection'><button className='nav-btn loggedIn-btn'>Show Favorites</button></Link> }
      { currentUser && home && <Link className='link' to='/'><button className='nav-btn'>Search Audiobooks</button></Link> }
      {currentUser && <button className='nav-btn loggedIn-btn' onClick={logoutUser}>Logout</button> }
      </section>
    </nav>
      {currentUser && home && <div className='all-genres'>{allGenres}</div>}
    </>
  )
}
export const mapStateToProps = state => ({
  genres: state.genres
})

export const mapDispatchToProps = dispatch => (
  bindActionCreators({ logoutUser, showGenreFilter, showAllFilter }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Nav);
