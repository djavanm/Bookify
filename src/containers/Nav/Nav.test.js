import React from 'react';
import { shallow } from 'enzyme';
import { Nav, mapStateToProps, mapDispatchToProps } from './Nav';
import { logoutUser, showGenreFilter, showAllFilter} from '../../actions';

describe('Nav', () => {
  let wrapper
  const currentUserMock = {
    name: 'Sam',
    password: 'blah',
    email: 'blah@gmail.com'
  }
  const logoutUserMock = jest.fn();
  const genresMock = ['Mystery', 'Comedy']
  const mockState = {
      currentBooks: [],
      currentUser: currentUserMock,
      favorites: [],
      genres: genresMock,
      genreFilter: '',
      searchFilter: { start: 0, end: 10, length: 10},
    };

  beforeEach(() => {
    wrapper = shallow(<Nav
      home={true}
      currentUser={currentUserMock}
      logoutUser={logoutUserMock}
      genres={genresMock}
      showGenreFilter={showGenreFilter}
      showAllFilter={showAllFilter}
       />)
  })

  it('should match the snapshot with the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('it calls dispatch with the showAll action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = showAllFilter();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.showAllFilter();

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('it calls dispatch with the showGenre action', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = showGenreFilter('Comedy');
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.showGenreFilter('Comedy');

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('mapStateToProps should grab the props it needs', () => {
    const expected = {
      genres: genresMock
    }
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});
