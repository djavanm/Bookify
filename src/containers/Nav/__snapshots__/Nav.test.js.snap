// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with the data passed through 1`] = `
<Fragment>
  <nav>
    <h1
      className="header-title"
    >
      Bookify 
      <FaLeaf
        color="#07804B"
        size={40}
      />
       
    </h1>
    <section
      className="nav-buttons"
    >
      <h2
        className="nav-welcome"
      >
         Welcome,
        <span
          className="user-name"
        >
           
          Sam
        </span>
          
      </h2>
      <Link
        className="link"
        to="/"
      >
        <button
          className="nav-btn"
        >
          Search Audiobooks
        </button>
      </Link>
      <button
        className="nav-btn loggedIn-btn"
        onClick={[MockFunction]}
      >
        Logout
      </button>
    </section>
  </nav>
  <div
    className="all-genres"
  >
    <button
      className="nav-btn"
      key="0"
      onClick={[Function]}
    >
      Show All
    </button>
    <button
      className="nav-btn"
      key="1"
      onClick={[Function]}
    >
      Mystery
    </button>
    <button
      className="nav-btn"
      key="2"
      onClick={[Function]}
    >
      Comedy
    </button>
  </div>
</Fragment>
`;
